game-core/src/main/java/games/strategy/engine/chat/HeadlessChat.java

```diff
@@ -44,25 +42,11 @@ public Chat getChat() {
   }
 
   @Override
-  public void updatePlayerList(final Collection<INode> players) {}
+  public void updatePlayerList(final Collection<ChatParticipant> players) {}
 
   @Override
   public void setChat(final Chat chat) {
-    if (this.chat != null) {
-      this.chat.removeChatListener(this);
-    }
-    this.chat = chat;
-    if (this.chat != null) {
-      this.chat.addChatListener(this);
-      synchronized (this.chat.getMutex()) {
-        allText = new StringBuilder();
-        for (final ChatMessage message : this.chat.getChatHistory()) {
-          addChatMessage(message.getMessage(), message.getFrom().getValue());
-        }
-      }
-    } else {
-      updatePlayerList(Collections.emptyList());
-    }
+    throw new UnsupportedOperationException("Headless bots do not support resetting of chat");
```

Comment:
Would you mind explaining the reasoning behind this change?

Reply:
This method is never called in a headless context, the exception is here so we can be notified in case I missed something.

It was a progressive step to see that this method did nothing:
- the history copy was always a copy of an empty list
- update player list is a no-op.

Seems we had a copy/paste of this method and maintained it "blindly"

Reply:
Would be great to move out of the interface, beyond the mutability aspect. It's risky to make too many changes, this is a pretty brittle are of the code, I've a bad taste in my mouth after spending 2 hours debugging a remote not found error before this was ready to submit. 

Only question I think is who follows up, let me know if that is a refactor you'd like to work on.

Reply:
Sure, I can create a PR for that